stock_account_product_run_fifo_hook/hooks.py

```diff
@@ -122,157 +125,181 @@ def _run_fifo_new(self, quantity, company):
-                candidates = all_candidates
-            if not candidates:
-                break
-            qty_to_take_on_candidates = abs(svl_to_vacuum.remaining_qty)
-            qty_taken_on_candidates = 0
-            tmp_value = 0
-            taken_data = {}
-            for candidate in candidates:
-                qty_taken_on_candidate = min(
-                    candidate.remaining_qty, qty_to_take_on_candidates
+
+        new_svl_vals_real_time = []
+        new_svl_vals_manual = []
+        real_time_svls_to_vacuum = ValuationLayer
+
+        for product in self:
+            all_candidates = all_candidates_by_product[product.id]
+            current_real_time_svls = ValuationLayer
+            for svl_to_vacuum in svls_to_vacuum_by_product[product.id]:
+                # We don't use search to avoid executing _flush_search and to decrease interaction with DB
+                candidates = all_candidates.filtered(
+                    lambda r: r.create_date > svl_to_vacuum.create_date
+                    or r.create_date == svl_to_vacuum.create_date
+                    and r.id > svl_to_vacuum.id
                 )
-                taken_data[candidate.id] = {"quantity": qty_taken_on_candidate}
-                qty_taken_on_candidates += qty_taken_on_candidate
+                if not candidates:
+                    break
+                qty_to_take_on_candidates = abs(svl_to_vacuum.remaining_qty)
+                qty_taken_on_candidates = 0
+                tmp_value = 0
+                taken_data = {}
+                for candidate in candidates:
+                    qty_taken_on_candidate = min(
+                        candidate.remaining_qty, qty_to_take_on_candidates
+                    )
+                    qty_taken_on_candidates += qty_taken_on_candidate
+
+                    taken_data[candidate.id] = {"quantity": qty_taken_on_candidate}
+
+                    candidate_unit_cost = (
+                        candidate.remaining_value / candidate.remaining_qty
+                    )
+                    value_taken_on_candidate = (
+                        qty_taken_on_candidate * candidate_unit_cost
+                    )
+                    value_taken_on_candidate = candidate.currency_id.round(
+                        value_taken_on_candidate
+                    )
+
+                    taken_data[candidate.id].update(
+                        {
+                            "value": value_taken_on_candidate,
+                        }
+                    )
+
+                    new_remaining_value = (
+                        candidate.remaining_value - value_taken_on_candidate
+                    )
+
+                    candidate_vals = {
+                        "remaining_qty": candidate.remaining_qty
+                        - qty_taken_on_candidate,
+                        "remaining_value": new_remaining_value,
+                    }
+                    # Start Hook
```

Comment:
@LoisRForgeFlow Here we add a function helper to allow change data to write in candidates

Reply:
Perfect, that's what I missed

Thanks!